src/TMS/dev/HeatPump.cpp

```diff
@@ -0,0 +1,36 @@
+#include <TMS/dev/HeatPump.hpp>
+
+namespace TMS {
+
+HeatPump::HeatPump(IO::PWM& pwm, uint8_t initSpeed) : pwm(pwm) {
+    // Ensures initSpeed falls in the required range
+    if (initSpeed < MIN_INIT_SPEED) {
+        this->initSpeed = MIN_INIT_SPEED;
+    } else if (initSpeed > MAX_SPEED) {
+        this->initSpeed = MAX_SPEED;
+    } else {
+        this->initSpeed = initSpeed;
+    }
+}
+
+void HeatPump::init() {
+    float initDutyCycle = SPEED_TO_DUTY_CYCLE(initSpeed);
+    pwm.setPeriod(MIN_INIT_TIME / initDutyCycle);
+    pwm.setDutyCycle(initDutyCycle);
+}
+
+void HeatPump::setSpeed(uint8_t speed) {
+    // Ensures speed falls in the required range
+    if (speed > MAX_SPEED) {
+        speed = MAX_SPEED;
+    }
+
+    float dutyCycle = SPEED_TO_DUTY_CYCLE(speed);
+    pwm.setDutyCycle(dutyCycle);
+}
```

Comment:
Do you apply the 3 ms pulse here when setting a speed after it was previously stopped?

Reply:
This is related to the minimum initialization speed. In the new implementation, the impulse is taken care of in this method. Let me know if that is clear now.